### PR TITLE
fix: [CI-6195]: Include scm for all delegates

### PR DESCRIPTION
**Pull request by smjt-h**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41417)

### CODE REVIEW

File: 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java [hunk 1]
```diff
@@ -48,6 +48,8 @@ public class InstallUtils {
   private static final String amd64 = "amd64";
   private static final String arm64 = "arm64";
 
+  List<ClientTool> internalTool = Arrays.asList(ClientTool.SCM);
```
Comment (aman-harness) [inline 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java:51]: since its a list, name should be internalTool

Comment (aman-harness) [reply 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java:51]: add private

Comment (aman-harness) [reply 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java:51]: better to name it as `internalToolsList`

File: dockerization/delegate/scripts/client_tools.sh [hunk 2]
```diff
@@ -42,4 +42,7 @@ if [[ ( -z "$DISABLE_CLIENT_TOOLS") || ("$DISABLE_CLIENT_TOOLS" = "false") ]]; t
   && curl -s -L -o client-tools/scm/9ddfb2b9/scm https://app.harness.io/public/shared/tools/scm/release/9ddfb2b9/bin/linux/$ARCH/scm
 else
   echo "Client tools are disabled"
+  echo "Installing client tools for scm"
```
Comment (aman-harness) [inline dockerization/delegate/scripts/client_tools.sh:45]: Client tools are disabled and Installing client tools for scm are opposite messages. 
We can simply print Installing scm binary

File: dockerization/delegate/scripts/client_tools.sh [hunk 3]
```diff
@@ -42,4 +42,7 @@ if [[ ( -z "$DISABLE_CLIENT_TOOLS") || ("$DISABLE_CLIENT_TOOLS" = "false") ]]; t
...
+  && mkdir -m 777 -p client-tools/scm/9ddfb2b9 \
```
Comment (aman-harness) [inline dockerization/delegate/scripts/client_tools.sh:46]: @smjt-h Why this starts with `&&`. This is a new statement. Should not start with &&

Comment (smjt-h) [reply dockerization/delegate/scripts/client_tools.sh]: fixed

File: dockerization/delegate/scripts/client_tools.sh (hunk 2 above)
Comment (smjt-h) [reply dockerization/delegate/scripts/client_tools.sh]: done

File: 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java (hunk 1 above)
Comment (smjt-h) [reply 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java]: done

**Comment by bharatgoelharness** [comment]
@smjt-h Please test all the scenarios and update the PR with your testing.

File: 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java [hunk 4]
```diff
@@ -193,6 +195,13 @@ private static Path installTool(
     }
   }
 
+  private static boolean isInternalTool(ClientTool tool) {
+    if (internalToolsList.contains(tool)) {
```
Comment (aman-harness) [inline 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java:199]: return internalToolsList.contains(tool) should suffice.

Comment (aman-harness) [reply 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java:199]: no if condition is required. Its kind of redundant code.

Comment (smjt-h) [reply 960-api-services/src/main/java/io/harness/delegate/clienttools/InstallUtils.java]: done

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41417#-:-NJJpkZb9xyV29JJOr1S:bnfp4nl)* status: :shipit: complete! ✓

**Review by gaurav-jain1 — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41417)* status: :shipit: complete! ✓

**Pull request merged**